library/ssl_tls13_client.c

```diff
@@ -1865,6 +1918,7 @@ int mbedtls_ssl_tls13_handshake_client_step( mbedtls_ssl_context *ssl )
 #if defined(MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE)
         case MBEDTLS_SSL_CLIENT_CCS_AFTER_SERVER_FINISHED:
         case MBEDTLS_SSL_CLIENT_CCS_BEFORE_2ND_CLIENT_HELLO:
+        case MBEDTLS_SSL_CLIENT_CCS_AFTER_CLIENT_HELLO:
```

Comment:
TLS 1.3 spec:
"If not offering early data, the client sends a dummy change_cipher_spec record (see the third paragraph of Section 5) immediately before its second flight. This may either be before its second ClientHello or before its encrypted handshake flight. If offering early data, the record is placed immediately after the first ClientHello."
As we are not offering early data yet, I do not think we have to add this state yet. See https://github.com/hannestschofenig/mbedtls/pull/385#discussion_r799275543 for additional information.